packages/loot-core/src/shared/schedules.ts

```diff
@@ -369,6 +369,24 @@ export function getUpcomingDays(upcomingLength = '7'): number {
       );
     }
     default:
+      if (upcomingLength.includes('-')) {
+        const [num, unit] = upcomingLength.split('-');
+        const value = Math.max(1, parseInt(num, 10));
+        switch (unit) {
+          case 'day':
+            return value;
+          case 'week':
+            return value * 7;
+          case 'month':
+            const future = monthUtils.addMonths(today, value);
+            return monthUtils.differenceInCalendarDays(future, today);
```

Comment:
Maybe this is the fix, it was the way "oneMonth" is done above.

```suggestion
            return monthUtils.differenceInCalendarDays(future, month);
```

Reply:
make that fix in month and year and seemed to solve the issue